parser/v2/templateparser.go

```diff
@@ -28,7 +28,7 @@ var templateExpressionParser = parse.Func(func(pi *parse.Input) (r templateExpre
 	// templ (x []string) Test() {
 	// becomes:
 	// func (x []string) Test() templ.Component {
-	if r.Expression, err = parseGoFuncDecl(pi); err != nil {
+	if _, r.Expression, err = parseTemplFuncDecl(pi); err != nil {
```

Comment:
I wonder if there's some debug value in the name of the templ, it could be passed down the context.

For example, the context could identify slow rendering components, or could print out a series of names: `Page > Body > content > heading`.

Not important, or that you need to do anything with this PR. Its just that your change sparked ideas. 😀